src/types.rs

```diff
@@ -475,3 +506,126 @@ impl<'buf> RecvMsgOut<'buf> {
         self.header.flags
     }
 }
+
+/// [CancelBuilder] constructs match criteria for request cancellation.
+///
+/// The [CancelBuilder] can be used to selectively cancel one or more requests
+/// by user_data, fd, fixed fd, or unconditionally.
+///
+/// ### Examples
+///
+/// ```
+/// use io_uring::types::{CancelBuilder, Fd, Fixed};
+///
+/// // Match all in-flight requests.
```

Comment:
Probably should read 'Match any ...'. It will just cancel the first one found. The `All` bit isn't set. Maybe given an any().all() doc example too.

Reply:
https://github.com/tokio-rs/io-uring/pull/225/files#diff-3cb81e12d8f6c5840aad0b94bd6ece48c86065b24019895ff7fcf43a34e3a919R118-R120

Hmm `CancelBuilder::any()` (even without `.all()`) does seem to cancel every request in the ring, not just the first one found. 

AFAICT `CancelBuilder::any()` and `CancelBuilder::any().all()` are equivalent. Unless I'm missing something I don't think there is a way to say "cancel any single request".

Reply:
Maybe we can say "this will cancel any and all in-flight requests"?

Reply:
My bad. If Any is already getting All, then I misunderstood, or mis-remembered, the man page. It never made sense to me that any would match just one, but I thought that's what they had implemented.